app/Exceptions/Handler.php

```diff
@@ -62,14 +62,14 @@ public function register(): void
         });
 
         $this->renderable(function (Exception $exception) {
-            if ($exception instanceof HttpException && $exception->getStatusCode() == 500) {
+            if ($exception instanceof HttpException && $exception->getStatusCode() >= 500) {
                 return $this->handle500Error($exception);
             }
         });
     }
 
     protected function handle500Error(Exception $exception): Response
     {
-        return Inertia::render('Error');
+        return Inertia::render('Error', ['statusCode' => '500']);
```

Comment:
```suggestion
        return Inertia::render('Error', ['statusCode' => 500]);
```

the status code can be a number